### PR TITLE
Colors & Text Formatting Implementation

### PR DESCRIPTION
**Pull request by OcelotWalrus**
This PR implements text colors as well as text formatting to the game engine.

**NOTE: At every end of print() functions, always add `+ color_reset_all` to make sure the text formats well. (If you don't do that, the formatting will continues on the below lines, even if they are other print() functions!**

I let you figure out what the text formatting/color pattern will be.

### CODE REVIEW

**Comment by OcelotWalrus** [comment]
I'll work on making ASCII thumbnails for each current item and make it so it's displayed in-game. I will make a markdown document about creating item thumbnails that you'll be able to put in the wiki!

**Comment by lumbar527** [comment]
Maybe if there was a
```
print("YOU DIED!")
```
message, it could be red.

If there is an item where you are, it would be said using green. Enemies where you are would be orange.

**Comment by lumbar527** [comment]
> I'll work on making ASCII thumbnails for each current item and make it so it's displayed in-game. I will make a markdown document about creating item thumbnails that you'll be able to put in the wiki! (In an other PR).

I was actually thinking that at some point I could have actual images for weapon infos using Pygame. And maybe that I could gradual move things over to have some more graphics.

**Comment by OcelotWalrus** [comment]
> I was actually thinking that at some point I could have actual images for weapon infos using Pygame. And maybe that I could gradual move things over to have some more graphics.

I don't really think it's a good idea. I like the fact that it's in the terminal. It kinda make it cool.

**Comment by OcelotWalrus** [comment]
> Maybe if there was a
> 
> ```
> print("YOU DIED!")
> ```
> 
> message, it could be red.
> 
> If there is an item where you are, it would be said using green. Enemies where you are would be orange.
> 
> If you have better ideas, go ahead and suggest them.

That's good for me. I'm adding that right now.

**Comment by lumbar527** [comment]
> > I was actually thinking that at some point I could have actual images for weapon infos using Pygame. And maybe that I could gradual move things over to have some more graphics.
> 
> I don't really think it's a good idea. I like the fact that it's in the terminal. It kinda make it cool.

Yeah, I guess. Thanks for working on them, then!

**Comment by OcelotWalrus** [comment]
> Yeah, I guess. Thanks for working on them, then!

You're welcome. You can check the colors changes at https://github.com/lumbar527/Dungeons-of-Kathallion/pull/7/commits/f8940a96695ff7d991b9e330916eda6c226b13a5.

**Comment by OcelotWalrus** [comment]
I cannot change the enemy color in fights because you currently use a module 'battle' for this. I think it would be better that you/we make our own function for battle so we can fully customize it.

**Comment by lumbar527** [comment]
There is a function: it's in `battle.py`. (I'm actually working on fixing something in it now)

**Comment by OcelotWalrus** [comment]
> There is a function: it's in `battle.py`. (I'm actually working on fixing something in it now)

Yes you're right, sorry I didn't read well.

**Comment by lumbar527** [comment]
It's fine. I'll just test the code and then probably merge.

**Comment by OcelotWalrus** [comment]
Thanks!

File: main.py [hunk 1]
```diff
@@ -45,7 +78,7 @@
     save_file = "save_" + delete_save + ".yaml"
     check_file = os.path.isfile(save_file)
     if check_file == False:
-        print("ERROR: Couldn't find save file '" + save_file + "'")
+        print(colore_red + color_style_bright + "ERROR: Couldn't find save file '" + save_file + "'" + color_reset_all)
```
Comment (lumbar527) [inline main.py]:
```suggestion
        print(color_red + color_style_bright + "ERROR: Couldn't find save file '" + save_file + "'" + color_reset_all)
```

File: main.py [hunk 2]
```diff
@@ -114,9 +147,10 @@ def run(play):
         map_location_x = search_specific_x()
         map_location_y = search_specific_y()
         print("Coordinates:")
-        print("X: " + str(map_location_x))
-        print("Y: " + str(map_location_y))
+        print(color_green + colore_style_bright + "X: " + color_reset_all + str(map_location_x))
+        print(color_green + colore_style_bright + "Y: " + colore_reset_all + str(map_location_y))
         print(" ")
+        print(color_green + colore_style_bright + "Possilbe actions:" + color_reset_all)
```
Comment (lumbar527) [inline main.py]:
```suggestion
        print(color_green + color_style_bright + "Possible actions:" + color_reset_all)
```

File: main.py [hunk 3]
```diff
@@ -92,17 +125,17 @@ def search_specific_y():
 
 # gameplay here:
 def run(play):
-    separator = "###############################"
+    separator = color_style_bright + "###############################" + color_reset_all
     print(separator)
-    print("Reserved keys:")
+    print(color_green + colore_style_bright + "Reserved keys:" + color_reset_all)
```
Comment (lumbar527) [inline main.py]:
```suggestion
    print(color_green + color_style_bright + "Reserved keys:" + color_reset_all)
```

File: main.py [hunk 4]
```diff
@@ -92,17 +125,17 @@ def search_specific_y():
...
     print("N: Go north")
     print("S: Go south")
     print("E: Go east")
     print("W: Go west")
     print("I: View items. When in this view, type the name of an item to examine it.")
     print("Q: Quit game")
     print(" ")
-    print("Hints:")
+    print(color_green + colore_style_bright +"Hints:" + colore_reset_all)
```
Comment (lumbar527) [inline main.py]:
```suggestion
    print(color_green + color_style_bright +"Hints:" + colore_reset_all)
```

File: main.py [hunk 5]
```diff
@@ -114,9 +147,10 @@ def run(play):
         map_location_x = search_specific_x()
         map_location_y = search_specific_y()
         print("Coordinates:")
-        print("X: " + str(map_location_x))
-        print("Y: " + str(map_location_y))
+        print(color_green + colore_style_bright + "X: " + color_reset_all + str(map_location_x))
+        print(color_green + colore_style_bright + "Y: " + colore_reset_all + str(map_location_y))
```
Comment (lumbar527) [inline main.py]:
```suggestion
        print(color_green + color_style_bright + "X: " + color_reset_all + str(map_location_x))
        print(color_green + color_style_bright + "Y: " + colore_reset_all + str(map_location_y))
```

**Comment by OcelotWalrus** [comment]
I fixed some errors.

**Review by lumbar527 — COMMENTED** [review]
~Pretty good~
Oops, meant to say cancel.

**Review by lumbar527 — APPROVED** [review]

**Comment by OcelotWalrus** [comment]
Do not merge it!

**Comment by OcelotWalrus** [comment]
sorry

**Comment by lumbar527** [comment]
No, it's fine. I've reverted it so you can fix stuff. You might have to make a new PR, sorry.

**Comment by OcelotWalrus** [comment]
Why did you closed the PR?

**Comment by lumbar527** [comment]
It automatically closed when I merged, now I can't find a button to reopen, sorry.

**Pull request merged**